Makefile

```diff
@@ -28,28 +32,92 @@ dependencies: vendor node_modules ## Setup the dependencies
 
 .php-version: .php-version.dist
 	cp .php-version.dist .php-version
-	(cd tests/Application && ln -sf ../../.php-version)
+
+php.ini: php.ini.dist
+	cp php.ini.dist php.ini
 
 vendor: composer.lock ## Install the PHP dependencies using composer
+ifdef GITHUB_ACTIONS
+	${COMPOSER} install --prefer-dist
+else
 	${COMPOSER} install --prefer-source
+endif
 
 composer.lock: composer.json
+ifdef GITHUB_ACTIONS
+	${COMPOSER} update --prefer-dist
+else
 	${COMPOSER} update --prefer-source
+endif
 
-yarn.install: tests/Application/yarn.lock
+yarn.install: ${APP_DIR}/yarn.lock
 
-tests/Application/yarn.lock:
-	${YARN} install
-	${YARN} build
+${APP_DIR}/yarn.lock:
+	ln -sf ${APP_DIR}/node_modules node_modules
+	cd ${APP_DIR} && ${YARN} install && ${YARN} build
+# No CSS and JS on this plugin yet
+#	${YARN} install
+#	${YARN} encore prod
 
-node_modules: tests/Application/node_modules ## Install the Node dependencies using yarn
+node_modules: ${APP_DIR}/node_modules ## Install the Node dependencies using yarn
 
-tests/Application/node_modules: yarn.install
+${APP_DIR}/node_modules: yarn.install
+
+###
+### TEST APPLICATION
+### ¯¯¯¯¯
+
+application: .php-version php.ini ${APP_DIR} setup_application ${APP_DIR}/docker-compose.yaml
+
+${APP_DIR}:
+	(${COMPOSER} create-project --prefer-dist --no-scripts --no-progress --no-install sylius/sylius-standard="${SYLIUS_VERSION}" ${APP_DIR})
+
+setup_application:
+	rm -f ${APP_DIR}/yarn.lock
+	(cd ${APP_DIR} && ${COMPOSER} config repositories.plugin '{"type": "path", "url": "../../"}')
+	(cd ${APP_DIR} && ${COMPOSER} config extra.symfony.allow-contrib true)
+	(cd ${APP_DIR} && ${COMPOSER} require --no-scripts --no-progress --no-install --no-update monsieurbiz/${PLUGIN_NAME}="*@dev")
+	$(MAKE) apply_dist
+	$(MAKE) ${APP_DIR}/.php-version
+	$(MAKE) ${APP_DIR}/php.ini
+	(cd ${APP_DIR} && ${COMPOSER} install)
+
+${APP_DIR}/docker-compose.yaml:
+	rm -f ${APP_DIR}/docker-compose.yml
+	rm -f ${APP_DIR}/docker-compose.yaml
+	cp docker-compose.yaml.dist ${APP_DIR}/docker-compose.yaml
+.PHONY: ${APP_DIR}/docker-compose.yaml
+
+${APP_DIR}/.php-version: .php-version
+	(cd ${APP_DIR} && ln -sf ../../.php-version)
+
+${APP_DIR}/php.ini: php.ini
+	(cd ${APP_DIR} && ln -sf ../../php.ini)
+
+apply_dist:
+	cp -Rv dist/* ${APP_DIR}
```

Comment:
Can we have symlinks instead? Otherwise it will lead to some development issues.